Core/KeyMap.cpp

```diff
@@ -110,6 +110,12 @@ void UpdateNativeMenuKeys() {
 			cancelKeys.push_back(hardcodedCancelKeys[i]);
 	}
 
+	// For DInput controllers on Windows. Doesn't clash with XInput because that uses BUTTON_X etc.
+#if PPSSPP_PLATFORM(WINDOWS) && !PPSSPP_PLATFORM(UWP)
+	confirmKeys.push_back(InputMapping(DEVICE_ID_PAD_0, NKCODE_BUTTON_2));
+	cancelKeys.push_back(InputMapping(DEVICE_ID_PAD_0, NKCODE_BUTTON_3));
```

Comment:
A lot of controllers have different opinions on which buttons are which - I'm definitely not a fan of hardcoding these specific ones.  Why do we need to do this and didn't before?

-[Unknown]

Reply:
I'm not a fan either, but these match with the "autodetect" mappings we use for DInput pads on Windows - which happen to match the PS4 controller I have.

I can't get it to navigate PPSSPP's menus with the PS4 controller without this. Does that work for you somehow?

But maybe we should just pick up from the bindings, at least on PC where you always have other options to navigate anyway...

Reply:
I can check later.  It used to work just fine, of course.  I'm sure there are some people who do couch gaming with dinput controllers on media pcs, still...

I thought this did use the bindings automatically, and I already have buttons mapped.  I used to use (and still have) a controller that used different keys and it used to work fine in PPSSPP as well - after mapping it once.  Forcing the wrong mappings seems bad.  It's like the same stupid mistake the worst console->PC ports have made.

-[Unknown]

Reply:
We've always had hardcoded bindings just for the PPSSPP menus so that you can always navigate even if you accidentally clear the configurable bindings - would be really screwed on the NVIDIA Shield for example without that.

It's possible I broke something though, let me check an older build.

Reply:
Actually we already have code to try to pick up extra confirm/cancel buttons from the bindings. So I think I did break something.

Reply:
Yes this was just me screwing up. Fixed in the new PR.